utils/src/test/scala/za/co/absa/enceladus/utils/types/TypedStructFieldSuite.scala

```diff
@@ -179,5 +177,37 @@ class TypedStructFieldSuite extends FunSuite {
     checkField(typed, fieldType, fail, fail, nullable, Seq(ValidationError(errMsg)))
   }
 
+  test("Byte type not nullable, with default defined as Double") {
+    val fieldType = LongType
+    val nullable = false
+    val field = createField(fieldType, nullable, Some(3,14))
+    val typed = TypedStructField(field)
+    val errMsg = "'(3,14)' cannot be cast to long"
+    val fail = Failure(new NumberFormatException(errMsg))
+    checkField(typed, fieldType, fail, fail, nullable, Seq(ValidationError(errMsg)))
+  }
+
+  test("Boolean type nullable, with default defined as Boolean") {
+    val fieldType = BooleanType
+    val nullable = false
+    val field = createField(fieldType, nullable, Some(true))
+    val typed = TypedStructField(field)
+    checkField(typed, fieldType, Success(Some(Some(true))), Success(Some(true)), nullable)
+  }
+
+  test("Float type nullable, with default defined as Long") {
+    val fieldType = FloatType
+    val nullable = false
+    val field = createField(fieldType, nullable, Some(1000L))
```

Comment:
Not sure how this works. Yes, you can put anything into `metadata` inside a schema when you already have it as a Scala object. But `StructType` schema is received as a JSON from Menas. It will lose the difference between `Int`, `Long` and `Double`, won't it?

Maybe better to test metadata processing by having examples of JSONs that could be received from Menas and check what types can be recognized after deserialization.

Reply:
From the JSON it's read in the form how JSON interpreter understands it. E.g. 3.14 as Double, -1 as Integer.
I run a standardization task with these defaults seemed working fine.
Now want to check if Menas accepts such schema JSON

Reply:
As actually tested/observed by @Zejnilovic in Menas it doesn't work well (Issue #799)